handle converting pr to/from draft

matt-fidd opened the pull request:
The assumptions made here are:

- if a PR is a draft, we don't need the `[WIP]` title prefix
- if a PR is a draft, it will always be considered WIP so needs the WIP label
- if a PR is converted from a draft, it should be considered ready for review, and should update as such
- if a PR is converted back to a draft, it should drop the prefix for consistency and revert back to the WIP label
- if a PR is a draft we don't check for failing CI

twk3 commented on src/handlers/pullRequest.ts:
```diff
@@ -36,4 +36,28 @@ export default (app: Probot) => {
+    const pr = new PullRequest(context);
+
+    let title = pr.data.title;
+    if (pr.wip) {
```
This is probably unnecessary, and might be confusing to the author. If it already had WIP, and converting it to draft made that go away, they might think the bot is confused.

twk3 commented on src/handlers/pullRequest.ts:
```diff
@@ -36,4 +36,28 @@ export default (app: Probot) => {
+
+    let title = pr.data.title;
+    if (pr.wip) {
+      title = title.substring(6);
```
Can we use a smarter regex replacement? It's likely what someone (those who use WIP in the title themsevles when they create PRs) will expand our wip regex  to match a few different wip prefixes. Like `WIP: ` but more specifically, to not require the whitespace. Or to have too much whitespace between the wip and title that they would expect trimmed off when we process it like this.

matt-fidd commented:
I've addressed your comments, not my finest work hardcoding a length in when it used a regex to match the bit I was removing anyway!

I've also extended the regex to match other common patterns that were previously handled in Trafico so as not to break any folks workflow.

twk3 reviewed (approved):
Great, thanks @matt-fidd !
